airbyte-server/src/main/java/io/airbyte/server/handlers/ConnectionsHandler.java

```diff
@@ -227,26 +232,229 @@ private Builder<String, Object> generateMetadata(final StandardSync standardSync
+      final AirbyteCatalog existingCatalog = CatalogConverter.toApi(sync.getCatalog());
+      final AirbyteCatalog patchedCatalog = toPatchedAirbyteCatalog(existingCatalog, patch.getSyncCatalog());
+      sync.setCatalog(CatalogConverter.toProtocol(patchedCatalog));
+    }
+
+    // update the sync's schedule using the patch's scheduleType and scheduleData. validations occur in
+    // the helper to ensure both fields
+    // make sense together.
+    if (patch.getScheduleType() != null) {
+      ConnectionScheduleHelper.populateSyncFromScheduleTypeAndData(sync, patch.getScheduleType(), patch.getScheduleData());
+    }
+
+    // the rest of the fields are straightforward to patch. If present in the patch, set the field.
+    // Otherwise, leave the field unchanged.
+
+    if (patch.getName() != null) {
+      sync.setName(patch.getName());
+    }
+
+    if (patch.getNamespaceDefinition() != null) {
+      sync.setNamespaceDefinition(Enums.convertTo(patch.getNamespaceDefinition(), NamespaceDefinitionType.class));
+    }
+
+    if (patch.getNamespaceFormat() != null) {
+      sync.setNamespaceFormat(patch.getNamespaceFormat());
+    }
+
+    if (patch.getPrefix() != null) {
+      sync.setPrefix(patch.getPrefix());
+    }
+
+    if (patch.getOperationIds() != null) {
+      sync.setOperationIds(patch.getOperationIds());
+    }
+
+    if (patch.getStatus() != null) {
+      sync.setStatus(ApiPojoConverters.toPersistenceStatus(patch.getStatus()));
+    }
+
+    if (patch.getSourceCatalogId() != null) {
+      sync.setSourceCatalogId(patch.getSourceCatalogId());
+    }
+
+    if (patch.getResourceRequirements() != null) {
+      sync.setResourceRequirements(ApiPojoConverters.resourceRequirementsToInternal(patch.getResourceRequirements()));
+    }
+  }
+
+  private void validateConnectionPatch(final WorkspaceHelper workspaceHelper, final StandardSync persistedSync, final ConnectionUpdate patch) {
+    // sanity check that we're updating the right connection
+    Preconditions.checkArgument(persistedSync.getConnectionId().equals(patch.getConnectionId()));
+
+    // make sure all operationIds belong to the same workspace as the connection
     ConnectionHelper.validateWorkspace(
-        workspaceHelper,
-        persistedSync.getSourceId(),
-        persistedSync.getDestinationId(),
-        new HashSet<>(connectionUpdate.getOperationIds()));
+        workspaceHelper, persistedSync.getSourceId(), persistedSync.getDestinationId(), patch.getOperationIds());
+
+    // make sure the incoming schedule update is sensible. Note that schedule details are further
+    // validated in ConnectionScheduleHelper, this just
+    // sanity checks that fields are populated when they should be.
+    Preconditions.checkArgument(
+        patch.getSchedule() == null,
+        "ConnectionUpdate should only make changes to the schedule by setting scheduleType and scheduleData. 'schedule' is no longer supported.");
+
+    if (patch.getScheduleType() == null) {
```

Comment:
@timroes @krishnaglick I wonder if this will pose any problems for the front-end? My validation here is assuming that requests that include `scheduleData` will always include `scheduleType` as well. I think this would be a good thing to enforce to remove any ambiguity in the update, but there might be existing front-end code that will fail the validation